src/components/menuBar/menu-bar.spec.js

```diff
@@ -279,6 +279,108 @@ describe('material.components.menuBar', function() {
           return el;
         }
       });
+
+      describe('nested menus - close menu with escape key', function() {
+        var menuBar, menus, subMenuOpen, ctrl;
+
+        it('opens consecutive nested menus', inject(function($mdConstant) {
+          menuBar = setup();
+          ctrl = menuBar.controller('mdMenuBar');
+          menus = menuBar[0].querySelectorAll('md-menu md-menu');
+
+          angular.element(document.body).append(menuBar);
+
+          expect(getNumberOfOpenMenus()).toBe(0);
+
+          // Open the menu-bar menu
+          ctrl.focusMenu(1);
+          ctrl.openFocusedMenu();
+          waitForMenuOpen();
+
+          expect(getNumberOfOpenMenus()).toBe(1);
+
+          // Open the first nested menu
+          openSubMenu(0);
+          waitForMenuOpen();
+          expect(getOpenSubMenu().text().trim()).toBe('Sub 1 - Content');
+
+          expect(getNumberOfOpenMenus()).toBe(2);
+
+          // Close the first nested menu with escape key
+          pressKey(getOpenSubMenu(), $mdConstant.KEY_CODE.ESCAPE);
+          waitForMenuClose();
+
+          // Just main menu should be visible
+          expect(getNumberOfOpenMenus()).toBe(1);
+
+          menuBar.remove();
+        }));
+
+        function openSubMenu(index) {
```

Comment:
These seem to be duplicated from the parent `describe` block? Except for `getNumberOfOpenMenus ()`.

Can this all be done within that same `describe` for `nested menus`?